### PR TITLE
Upgrade dependencies and otel vuln fix 

### PR DESCRIPTION
**Pull request by cerberus20**
Bump  otelhttp/grpc , custom metrics apiserver. and k8s dependencies. Also upgrade go version in makefile

### CODE REVIEW

**Comment by jaybooth4** [comment]
Thanks so much for the CL [cerberus20](https://github.com/cerberus20)!

Should we also update the go version as part of this CL? https://github.com/kubernetes-sigs/prometheus-adapter/blob/master/Makefile#L11

**Comment by cerberus20** [comment]
yeah, updated @jaybooth4

**Comment by jaybooth4** [comment]
Nit @cerberus20 could we update go version to 1.22.2 to include the latest HTTP fixes?
https://go.dev/doc/devel/release#go1.22.0

**Comment by jaybooth4** [comment]
@dgrisonnet @olivierlemasle can we possibly get a quick review for these changes to improve security/vulns? Thanks!

**Comment by CatherineF-dev** [comment]
LGTM to trigger tests.

**Comment by dgrisonnet** [comment]
@cerberus20 could you please run `make update-lint`, it should make the verify test happy.

**Comment by dgrisonnet** [comment]
FWIW, prometheus-adapter is not impacted by the otel CVE

**Comment by dgrisonnet** [comment]
and I will most likely wait for https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/168 before cutting a new release of prometheus-adapter

**Comment by cerberus20** [comment]
@dgrisonnet I had updated linter version to 1.57.2, and verify lint passed on my local.  Hence `make update-lint` doesn't have any changes/errors on machine, what should be do here? Changing to 1.53.2 does produce some errors on verify lint

**Comment by cerberus20** [comment]
@dgrisonnet @CatherineF-dev Can we please retrigger failed test? thank you

File: go.mod [hunk 1]
```diff
@@ -4,112 +4,112 @@ go 1.20
...
-	sigs.k8s.io/custom-metrics-apiserver v1.27.0
+	k8s.io/api v0.28.1
+	k8s.io/apimachinery v0.28.1
+	k8s.io/apiserver v0.28.1
+	k8s.io/client-go v0.28.1
+	k8s.io/component-base v0.28.1
+	k8s.io/klog/v2 v2.100.1
+	k8s.io/kube-openapi v0.0.0-20230901164831-6c774f458599
+	k8s.io/metrics v0.28.1
+	sigs.k8s.io/custom-metrics-apiserver v1.28.0
```
Comment (CatherineF-dev) [inline go.mod:23]: Need to use latest custom-metrics-apiserver, https://github.com/kubernetes-sigs/prometheus-adapter/pull/655

Comment (cerberus20) [reply go.mod]: @CatherineF-dev thanks for letting me know, updated the version

Comment (cerberus20) [reply go.mod]: @CatherineF-dev  @dgrisonnet Could you please help  why verify lint test is failing? I ran the verify lint command on my local and it passes fine

image
image

Comment (cerberus20) [reply go.mod]: I have reverted the GOLANGCI_VERSION for now

File: cmd/adapter/adapter.go [hunk 2]
```diff
@@ -320,12 +318,6 @@ func main() {
 		cmd.OpenAPIConfig.Info.Version = "1.0.0"
 	}
 
-	if cmd.OpenAPIV3Config == nil && utilfeature.DefaultFeatureGate.Enabled(features.OpenAPIV3) {
```
[the remark this answers is not in the document]
Comment (cerberus20) [reply cmd/adapter/adapter.go:323]: ack, removed the feature check for OpenAPIV3. thanks

[the remark this answers is not in the document]
Comment (CatherineF-dev) [reply cmd/adapter/adapter.go:323]: I think we can use DefaultOpenAPIConfig as a temporary solution.
```
		cmd.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(generatedopenapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(api.Scheme, customexternalmetrics.Scheme))
```

1. Read more documentations, and find OpenAPIV3 is available in k8s 1.27+
2. This adapter should work for k8s 1.27- as well.

Was thinking a way to make it work for both 1.27+ and 1.27-. But haven't found it.

**Comment by chc5** [comment]
Any updates here? I'd like to get these upgrades submitted to address vulnerabilities in this project.

**Comment by dgrisonnet** [comment]
Superseded by https://github.com/kubernetes-sigs/prometheus-adapter/pull/659

**Pull request closed without merging**